### DISCUSSION TITLE
Introduction to Mito Engine

### DISCUSSION THREAD

**Discussion by leaf-potato** [GreptimeTeam/greptimedb#4498] [Q&A]
I am particularly interested in the implementation of the mito2 engine and would like to ask if there is any detailed design introduction information about mito2. The official website document of significant optimizations for handling time-series data scenarios is relatively brief: 
<img width="1519" alt="image" src="https://github.com/user-attachments/assets/a955fb22-93e3-4273-a2a2-afdb91076997">

**Comment by evenyag** [comment]
The engine is under active development and some codes and designs change frequently. Detailed documents may soon be outdated so we may not provide it currently.

Here are some diagrams about the engine internals.
https://greptimedb.rs/mito2/docs/index.html

**Reply by leaf-potato** [reply]
@evenyag Also, I would like to ask if there are any functions that the community needs to implement for the mito engine? I am very interested.
